src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/EvidenceSectionController.java

```diff
@@ -0,0 +1,284 @@
+
+  private final LookupService lookupService;
+
+  private final EvidenceUploadValidator evidenceUploadValidator;
+
+  private final EvidenceMapper evidenceMapper;
+
+
+  /**
+   * Handles the GET request for the evidence upload screen.
+   *
+   * @param activeCase         The details of the active case.
+   * @param model              The model for the view.
+   * @return The view name for the evidence upload view.
+   */
+  @GetMapping("/application/summary/evidence")
+  public String viewEvidenceRequired(
+      @SessionAttribute(ACTIVE_CASE) final ActiveCase activeCase,
+      final Model model) {
+
+    // Get the list of required evidence docs for this application.
+    final Mono<List<EvidenceDocumentTypeLookupValueDetail>> evidenceRequiredMono =
+        evidenceService.getDocumentsRequired(
+            String.valueOf(activeCase.getApplicationId()),
+            activeCase.getCaseReferenceNumber(),
+            activeCase.getProviderId());
+
+    // Retrieve the list of previously uploaded documents.
+    final Mono<EvidenceDocumentDetails> evidenceUploadedMono =
+        evidenceService.getEvidenceDocumentsForCase(
+            activeCase.getCaseReferenceNumber(),
+            APPLICATION);
+
+    Tuple2<List<EvidenceDocumentTypeLookupValueDetail>,
+        EvidenceDocumentDetails> combinedResult = Mono.zip(
+            evidenceRequiredMono,
+            evidenceUploadedMono)
+        .blockOptional()
+        .orElseThrow(() -> new CaabApplicationException("Failed to retrieve evidence data"));
+
+    List<EvidenceDocumentTypeLookupValueDetail> evidenceRequiredLookups = combinedResult.getT1();
+    List<BaseEvidenceDocumentDetail> evidenceUploaded = combinedResult.getT2().getContent();
+
+    final List<EvidenceRequired> evidenceRequired =
+        evidenceMapper.toEvidenceRequiredList(evidenceRequiredLookups, evidenceUploaded);
+
+    model.addAttribute(EVIDENCE_REQUIRED, evidenceRequired);
+    model.addAttribute("evidenceUploaded", evidenceUploaded);
+
+    return "application/summary/evidence-section";
+  }
+
+  /**
+   * Handles the GET request for the add evidence screen.
+   *
+   * @param activeCase Basic details of the active case.
+   * @param userDetail The user details.
+   * @param evidenceRequired List of evidence required.
+   * @param model              The model for the view.
+   * @return The view name for the evidence upload view.
+   */
+  @GetMapping("/application/evidence/add")
+  public String viewEvidenceUpload(
+      @SessionAttribute(ACTIVE_CASE) final ActiveCase activeCase,
+      @SessionAttribute(USER_DETAILS) final UserDetail userDetail,
+      @SessionAttribute(EVIDENCE_REQUIRED)
+      final List<EvidenceRequired> evidenceRequired,
+      final Model model) {
+
+    // Initialise the form data object.
+    final EvidenceUploadFormData evidenceUploadFormData = new EvidenceUploadFormData();
+    evidenceUploadFormData.setApplicationOrOutcomeId(activeCase.getApplicationId());
```

Comment:
If we start using  application id in the active case session variable, it might be worth raising a bug ticket to remove the standalone applicationid session attribute and use this moving forward

Reply:
I had meant to discuss the changes to ActiveCase with you a while ago. I didn't want to be retrieving the full Application details when all I needed was the basic id/case ref/providerId.
We should have a chat about this to make sure it's not getting a bit messy.